### ISSUE TITLE
When Catching Exception Still Displays Error

### ISSUE THREAD

**Issue by tanifgit** [intersystems-ru/webterminal#101]
version 2017.1 / CWTv4.2.15
It seems that if code includes a Try/Catch scope then if the exception occurs the Web Terminal still displays the error even if it is caught and handled.

For example -
Assume this simple class:

```COS
Class User.Catch
{

ClassMethod try()
{
	Try {
		Write "Trying...",!
		Set a=1/0
	} Catch {
		Write "Caught",!	
	}
}

}
```

Running this in Web Terminal will yield:

```
TESTDB > do ##class(User.Catch).try()
Trying...                               
Caught
Bye

<DIVIDE>ztry+3^User.Catch.1
║ Try {
║       Write "Trying...",!
╠       Set a=1/0
║ } Catch {
║       Write "Caught",!
```

I would argue that displaying the error in this case might be misleading, assuming for example this was intended logic.
Possibly it makes sense to add a flag whether to display the error in these kinds of cases or not.

For example the output in the "regular" Terminal would simply be -

```
TESTDB>do ##class(User.Catch).try()
Trying...
Caught
Bye
 
TESTDB>
```

**Comment by ZitRos** [comment]
Thank you for reporting this issue! I agree, this must be fixed.

The problem is in how WebTerminal Core class (`WebTerminal.Core`, class method `Loop`) handles the exceptions:

```COS
set $ZERROR = ""
try {
  xecute ..WaitCommand()
  use $io::"^" _ $ZName
} catch {}
// ...
do ..Something($ZERROR)
```

It just tests `$ZERROR` after execution if it has any value. I will try to figure out how to fix this behavior. By the way, any ideas are welcome.

**Comment by ZitRos** [comment]
It looks like putting `do ..Something($ZERROR)` inside try/catch block fixes this:

```COS
set $ZERROR = ""
try {
  xecute ..WaitCommand()
  use $io::"^" _ $ZName
} catch error {
  do ..Something(error)
}
// ...
```

![2017-06-14_111916](https://user-images.githubusercontent.com/4989256/27122431-54801430-50f3-11e7-889f-b58665309670.png)

But I need to ensure that all other cases will work as expected as well.

**Comment by eduard93** [comment]
> do ..Something(error) // THIS
Maybe `do ..Something($ZERROR)`

**Comment by ZitRos** [comment]
Thank you, Eduard. Makes sense.

The most important point here is to be sure that try/catch will always catch any kind of errors. I believe that I left catch block empty (in `WebTerminal.Core` class) for some reason...

**Comment by ZitRos** [comment]
Fixed in CWT v4.3.0. Thanks.
